src/RequestProcessor/Encoder/EncodingService.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System.Text;
+
+namespace RequestProcessor.Services.Encoder
+{
+    public class EncodingService : IEncodingService
+    {
+        public string DecodeBase64(string content)
+        {
+            if (content == null)
+            {
+                throw new ArgumentNullException(nameof(content));
+            };
+
+            Span<byte> buffer = new(new byte[content.Length]);
+            var isBase64 = Convert.TryFromBase64String(content, buffer, out int bytesWritten);
+            if (isBase64)
+            {
+                var s = Encoding.UTF8.GetString(buffer[..bytesWritten]);
```

Comment:
Why not just return on this line?

Reply:
Will do this in a subsequent change - I suspect it looks like this so someone can see the value more easily in a debugger.